### PR TITLE
feat: add COSE-C

### PR DESCRIPTION
**Pull request by gocarlos**
- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cose-c/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class CoseCStackConan(ConanFile):
+    name = "COSE-C"
```
Comment (uilianries) [inline recipes/cose-c/all/conanfile.py:7]: Use only lower-case for packages, even if the original package name is UPPER CASE or CamelCase

**Comment by gocarlos** [comment]
> Failure in build 4 (`19e5e3fb84255334e7efe23d20ac91463b0eeddd`):
> cose-c/20200326:
> 
> * **Error processing recipe: Linux x86_64, Release, gcc 5, libstdc++ . Options:  cose-c:shared-True**
>   You are depending on 'cn-cbor/1.0.0' but it is not in the repository

waiting for cn-cbor/1.0.0 e.g. #1313

**Comment by gocarlos** [comment]
> Some configurations of 'cose-c/20200417' failed in build 11 (`9515fc00cfdd918ed59ad375ac0d7900ba92b43f`):
> 
> * **Linux x86_64, Release, gcc 5, libstdc++ . Options:  cose-c:shared-True**
>   
>   * [Log download](https://c3i.jfrog.io/c3i/misc/logs/pr/1314/11/cose-c/20200417/1fb4b6ea872fa8b9d320eb1ac007b4882971adfb/create_stdout.txt)
>   * [Log error Download](https://c3i.jfrog.io/c3i/misc/logs/pr/1314/11/cose-c/20200417/1fb4b6ea872fa8b9d320eb1ac007b4882971adfb/create_stderr.txt)
>   * [Profile](https://c3i.jfrog.io/c3i/misc/logs/pr/1314/11/cose-c/20200417/1fb4b6ea872fa8b9d320eb1ac007b4882971adfb/profile.txt)
> * [Access to all the logs](https://c3i.jfrog.io/c3i/misc/logs/pr/1314/11/cose-c/20200417/)

needs https://github.com/cose-wg/COSE-C/pull/79

File: recipes/cose-c/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        self.requires("cn-cbor/1.0.0")
+
+        if self.options.use_mbedtls:
+            self.requires("mbedtls/2.16.3-gpl")
+        else:
+            self.requires("openssl/1.1.1d")
```
Comment (uilianries) [inline recipes/cose-c/all/conanfile.py:43]:
```suggestion
            self.requires("openssl/1.1.1f")
```

File: recipes/cose-c/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    def configure(self):
+        if self.settings.os in ["Windows", "Macos", "Android"]:
+            raise ConanInvalidConfiguration("cose-c does not support your OS at the moment")
```
Comment (uilianries) [inline recipes/cose-c/all/conanfile.py:47]: The upstream informs the error, but it's related to the compiler, not by OS. I think is better following the same behavior: https://github.com/cose-wg/COSE-C/blob/master/CMakeLists.txt#L112

**Comment by SSE4** [comment]
seems like it doesn't link with OpenSSL on macOS, need to investigate/debug locally

File: recipes/cose-c/all/conandata.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "20200430":
```
Comment (prince-chrismc) [inline recipes/cose-c/all/conandata.yml:2]:
```suggestion
  "cci.20200430":
```

File: recipes/cose-c/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("cn-cbor/1.0.0")
+
+        if self.options.use_mbedtls:
+            self.requires("mbedtls/2.16.3-gpl")
+        else:
+            self.requires("openssl/1.1.1f")
```
Comment (prince-chrismc) [inline recipes/cose-c/all/conanfile.py:47]:
```suggestion
            self.requires("openssl/1.1.1h")
```

File: recipes/cose-c/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (prince-chrismc) [inline recipes/cose-c/all/test_package/CMakeLists.txt:8]: We could test for the cmake targets https://github.com/cose-wg/COSE-C#building

**Comment by prince-chrismc** [comment]
> seems like it doesn't link with OpenSSL on macOS, need to investigate/debug locally

I suspect this might be the usually issue where by the standard snake variable is not set by conan cmake find package generator 

https://github.com/cose-wg/COSE-C/blob/97d1805e71b7a6770093c5e6790d46611680d563/src/CMakeLists.txt#L70

**Pull request closed without merging**